### PR TITLE
Throw ArgumentException when path contains invalid characters

### LINKED ISSUE #367: MockFileSystem.Directory.GetFiles throws DirectoryNotFoundException with invalid characters in directory path
When calling `MockFileSystem.Directory.GetFiles(string path, string searchPattern)` with a path containing invalid path characters from `MockFileSystem.Path.GetInvalidPathChars()` a `DirectoryNotFoundException` is thrown instead of the anticipated `ArgumentException` which gets thrown by `FileSystem.Directory.GetFiles(string path, string searchPattern)` creating a different behaviour between test and regular implementation.

### PR DESCRIPTION
**Pull request by robkeim**
Fixes #367

### CODE REVIEW

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs [hunk 1]
```diff
@@ -277,5 +277,20 @@ public void MockFileSystem_AddDirectory_TrailingSlashAllowedButNotRequired(strin
 
             Assert.IsTrue(fileSystem.FileExists(path2));
         }
+
+        [Test]
+        public void MockFileSystem_GetFiles_ThrowsCorrectExceptionForInvalidCharacters()
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs]: I prefer to be specific in the naming, so what about that:

```suggestion
        public void MockFileSystem_GetFiles_ThrowsArgumentExceptionForInvalidCharacters()
```

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs [hunk 2]
```diff
@@ -277,5 +277,20 @@ public void MockFileSystem_AddDirectory_TrailingSlashAllowedButNotRequired(strin
...
+        {
+            // Arrange
+            const string path = @"c:\";
+            var fileSystem = new MockFileSystem();
+            fileSystem.AddDirectory(XFS.Path(path));
+            
+            // Act
+            TestDelegate wrapped = () => fileSystem.Directory.GetFiles($"{path}{'\0'}.txt");
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs]: I don't like the name `wrapped` - maybe something like `getFilesWithInvalidCharacterInPath`?

File: System.IO.Abstractions.TestingHelpers/MockDirectory.cs [hunk 3]
```diff
@@ -176,8 +175,15 @@ public override string[] GetFiles(string path, string searchPattern)
 
         public override string[] GetFiles(string path, string searchPattern, SearchOption searchOption)
         {
-            if(path == null)
+            if (path == null)
+            {
                 throw new ArgumentNullException();
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers/MockDirectory.cs]: Please add argument name

File: System.IO.Abstractions.TestingHelpers/MockDirectory.cs [hunk 4]
```diff
@@ -176,8 +175,15 @@ public override string[] GetFiles(string path, string searchPattern)
...
+            }
+            
+            if (path.Any(c => Path.GetInvalidPathChars().Contains(c)))
+            {
+                throw new ArgumentException();
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers/MockDirectory.cs]: Please add argument name and message.

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs (hunk 1 above)
Comment (robkeim) [reply System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs]: Makes sense

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs (hunk 2 above)
Comment (robkeim) [reply System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs]: Changed

File: System.IO.Abstractions.TestingHelpers/MockDirectory.cs (hunk 3 above)
Comment (robkeim) [reply System.IO.Abstractions.TestingHelpers/MockDirectory.cs]: Done

File: System.IO.Abstractions.TestingHelpers/MockDirectory.cs (hunk 4 above)
Comment (robkeim) [reply System.IO.Abstractions.TestingHelpers/MockDirectory.cs]: Done

**Comment by robkeim** [comment]
Thanks @fgreinacher for the review. I integrated your feedback and pushed a new commit so you should be able to review again.

**Comment by fgreinacher** [comment]
Thanks, looks great now!

**Pull request merged**